components/shift_reg/README.md

```diff
@@ -0,0 +1,60 @@
+# `Shift Register`
+This driver can be used as an interface with a shift register in (still in progress) and out (such as [74HC595](https://www.ti.com/lit/ds/symlink/sn74hc595.pdf)).
+
+## Usage
+To use the library, it needs to be configured before initialize. To do so a `shift_reg_config_t` needs to be set.
+
+### Configuration
+The `shift_reg_config_t` struct store all necessaries configurtions and value of aech register.
+
+* `uint8_t num_reg` - Number of registers which will be used
+* `uint8_t *reg_value` - Vector for the last value of all registers; it can be used for know what is the actual value of the registers
+
+### Mode configuration
+* `struct mode`
+    * `shift_reg_dir_t dir` - Direction mode of the shift register
+    * `shift_reg_bit_mode_t bit_mode` - Bit mode
+
+### Pin configurations
+* `struct pin`
+    * `gpio_num_t clk` - Clock pin
+    * `gpio_num_t data` - Data/Signal pin
+    * `gpio_num_t latch` - Latch pin
+
+#### Direcition mode
+The `shift_reg_dir_t` says the direction of the register.
+* `SHIFT_DIR_OUTPUT` - Set the register as output
+* `SHIFT_DIR_INPUT` - Set the register as input
+* `SHIFT_DIR_INPUT_OUTPUT` - Set the register as output and input
+
+#### First bit configuration
+The `shift_reg_bit_mode_t` says the bit mode of the data.
+* `SHIFT_BIT_MODE_LSB` - Start send data from the lowest significant bit
+* `SHIFT_BIT_MODE_MSB` - Start send data from the most significandt bit
+
+### Initialization
+To initialize the shift register it is going to need to call the `esp_err_t shift_reg_init(shift_reg_config_t *dev)`, by passing the created shift register configurations.
+
+### De-initialization
+Since the `uint8_t *reg_value` is created accordingly of the number of registers, the vector allocate the necessary size in the heap memory, so `esp_err_t shift_reg_deinit(shift_reg_config_t *dev)` can be used to free this memory.
```

Comment:
this should go to the header file so that users can see this in the official documentation. see other documentation pages under `docs/source/groups`.